frontend/cypress/integration/common/sidecar_injection.ts

```diff
@@ -0,0 +1,274 @@
+    cy.get('[data-test=overview-type-LIST]').click();
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).click();
+    cy.get(`[data-test=${enabledOrDisabled}-${this.targetNamespace}-namespace-sidecar-injection]`).click();
+    cy.intercept(Cypress.config('baseUrl') + '/api/namespaces').as('namespacesReload');
+    cy.get('[data-test=confirm-traffic-policies]').click();
+    cy.wait('@namespacesReload')
+});
+
+When('I remove override configuration for sidecar injection in the namespace', function () {
+    cy.visit('/console/overview');
+    cy.get('[data-test=overview-type-LIST]').click();
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).click();
+    cy.get(`[data-test=remove-${this.targetNamespace}-namespace-sidecar-injection]`).click();
+    cy.intercept(Cypress.config('baseUrl') + '/api/namespaces').as('namespacesReload');
+    cy.get('[data-test=confirm-traffic-policies]').click();
+    cy.wait('@namespacesReload')
+});
+
+function switchWorkloadSidecarInjection(enableOrDisable) {
+    cy.visit(`/console/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}`);
+    cy.intercept(Cypress.config('baseUrl') + `/api/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}**`).as('workloadReload');
+    cy.get('[data-test="workload-actions-dropdown"] button').click();
+    cy.get(`button[data-test=${enableOrDisable}_auto_injection]`).click();
+    cy.wait('@workloadReload');
+}
+
+When('I override the default policy for automatic sidecar injection in the workload to {string} it', switchWorkloadSidecarInjection);
+
+When('I change the override configuration for automatic sidecar injection in the workload to {string} it', switchWorkloadSidecarInjection);
+
+When('I remove override configuration for sidecar injection in the workload', function () {
+    switchWorkloadSidecarInjection.apply(this, ['remove']);
+});
+
+Then('I should see the override annotation for sidecar injection in the namespace as {string}', function (enabled) {
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}]`).contains(`istio-injection: ${enabled}`).should('exist');
+});
+
+Then('I should see no override annotation for sidecar injection in the namespace', function () {
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}]`).contains(`istio-injection`).should('not.exist');
+});
+
+Then('the workload should get a sidecar', function () {
+    cy.get('[data-test=missing-sidecar-badge-for-sleep-workload-in-default-namespace]').should('not.exist');
+});
+
+Then('the sidecar of the workload should vanish', function () {
+    cy.get('[data-test=missing-sidecar-badge-for-sleep-workload-in-default-namespace]').should('exist');
+});
+
+Then('I should see no override annotation for sidecar injection in the workload', function () {
+    cy.get('#WorkloadDescriptionCard').then($card => {
+        if ($card.find('label_more').length) {
+            cy.wrap($card).get('label_more').click();
+        }
```

Comment:
There's some warnings about conditional testing. Not sure if that applies here or not: https://docs.cypress.io/guides/core-concepts/conditional-testing

Reply:
I share this concern. Indeed, I stumbled on that guide when I was finding a way to implement this specific test.

May I ask you for your help here? This is the best way I could think of to implement this. I did add a few `cy.intercept` together with a `cy.wait` to try to prevent a flaky test. But I'm not sure if that will be enough. So, suggestions are welcome!

Reply:
You can use the pod's tooltip to get the list of annotations. That can save to find the "label_more", "click" and wait.

Reply:
Looking at this again and what it's trying to test. I think it's fine to click the "show more" labels button so that you can see all the labels before querying for a specific one. You can also get them from the tooltip as Lucas suggested above if that's easier.

Reply:
@lucasponce suggestion sounded quite good to remove the conditional. However, I found that the test becomes flaky, depending on how quickly the pod re-creates. It sometimes happens that there are two pods listed: the one that is terminating and the one that is being created.